services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -16,30 +18,60 @@
 _logger = logging.getLogger(__name__)
 
 
+_PAST_X_MINUTES = 30  # in promql query the first part: [30m:1m]
+_RESOLUTION_MINUTES = 1  # in promql query the second part: [30m:1m]
+_PROME_QUERY_PARAMS_TIMEDELTA_MINUTES = (
+    25  # should be less than _PAST_X_MINUTES (so there is small overlap)
+)
+
+
+class _PromQueryParameters(BaseModel):
+    image_regex: str
+    scrape_timestamp: datetime
+
+    class Config:
+        validation = False
+
+
 def _prometheus_sync_client_custom_query(
-    prometheus_client: PrometheusConnect, promql_cpu_query: str
+    prometheus_client: PrometheusConnect,
+    promql_cpu_query: str,
+    scrape_timestamp: datetime,
 ) -> list[dict]:
-    _logger.info("Querying prometheus with: %s", promql_cpu_query)
-    data: list[dict] = prometheus_client.custom_query(promql_cpu_query)
+    rfc3339_str = scrape_timestamp.isoformat("T")
```

Comment:
MINOR: I suggest using annotations intead of adding type to the variable name ``rfc3339: str = scrape_timestamp.isoformat("T")``